adaptors task/student/include/impl.h

```diff
@@ -11,4 +13,48 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isValid(const std::string& source)
+{
+   std::stack<char> charsStack;
+   bool result = true;
+
+   auto checkCloseBrackets{[&charsStack](const char ch){
+         if (!charsStack.empty() && ch == charsStack.top())
+         {
+            charsStack.pop();
+            return true;
+         }
+         else
+         {
+            return false;
+         }
+      }};
+
+   for (const char& ch : source)
+   {
+      switch (ch) {
+      case '(':
```

Comment:
USE "brackets" VECTOR FROM .cpp FILE! It is mandatory!

Reply:
Already reworked. Please check.